packages/components/src/utils/css-variables.tsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { VisuallyHidden } from '../visually-hidden';
+
+type CSSVariableReplacerProps = {
+	cssString?: string | null;
+	/**
+	 * Callback to be called when CSS variables in the given string
+	 * have been replaced with their computed values.
+	 *
+	 * Should be memoized to avoid unnecessary reflows.
+	 */
+	onChange: ( args: {
+		replacedCssString: string;
+		computedVariables: ComputedCSSVariables;
+	} ) => void;
+};
+type ComputedCSSVariables = Record< string, string >;
+type VarFunction = {
+	/** Start index of match. */
+	start: number;
+	/** End index of match. */
+	end: number;
+	/** Matched string, e.g. `var( --foo, #000 )`. */
+	raw: string;
+	/** CSS variable name, e.g. `--foo`. */
+	value: string;
+	/** Second argument of the `var()`, which could be a literal or another `var()`. */
+	fallback?: string;
+};
+
+/**
+ * Find the index of the matching closing parenthesis for a given opening parenthesis in a string.
+ */
+function findMatchingParenthesis( str: string, startPos: number ) {
+	let stack = 0;
+
+	for ( let i = startPos; i < str.length; i++ ) {
+		if ( str[ i ] === '(' ) {
+			stack++;
+		} else if ( str[ i ] === ')' ) {
+			stack--;
+			if ( stack === 0 ) {
+				return i;
+			}
+		}
+	}
+
+	throw new Error( 'No matching closing parenthesis found.' );
+}
+
+/**
+ * Find all `var()` functions in a CSS string.
+ */
+export function findVarFunctionsInString( str: string ) {
+	const regex = /(?<=\bvar)\(/g;
+	const matches: VarFunction[] = [];
+
+	let openingParen;
+	while ( ( openingParen = regex.exec( str ) ) !== null ) {
+		const closingParen = findMatchingParenthesis( str, openingParen.index );
+		const [ start, end ] = [
+			openingParen.index - 'var'.length,
+			closingParen + 1,
+		];
+		const raw = str.slice( start, end );
+		const value = raw.match( /--[\w-]+/ )?.[ 0 ];
+
+		if ( ! value ) {
+			throw new Error( 'No CSS variable found in var() function.' );
+		}
+
+		matches.push( {
+			start,
+			end,
+			raw,
+			value,
+			fallback: raw.match( /,(.+)\)/ )?.[ 1 ].trim(),
```

Comment:
Same about this regex and the rest of the regexes in here. In addition to the small performance benefit, it can actually make the code more readable if regexes are declared as constants and referenced by name.